changelogs/unreleased/gh-10361-box-zero-len-mp-decoding.md

```diff
@@ -0,0 +1,3 @@
+## bugfix/box
+
+* Fixed reading after MsgPack end of incorrect zero length tuple (gh-10361).
```

Comment:
I don't like the change log because it's unclear what "zero length tuple" or "zero length datetime" is. Better say that we fixed a crash on an attempt to decode a malformed MsgPack extension with zero payload.

Also, I think it's enough to have one bugfix/core changelog entry for both issues (because essentially it's the same issue) and better squash the two commits in one.

Reply:
Fixed.